app/controllers/tcsp/SummaryController.scala

```diff
@@ -37,11 +38,29 @@ class SummaryController @Inject()
   val statusService: StatusService
 ) extends BaseController {
 
+  def sortProviders(data: Tcsp): List[String] = {
+
+    val sortedList = for {
+      types <- data.tcspTypes
+      providers <- Some(types.serviceProviders)
+      labels <- Some(providers.collect {
+          case provider if provider.value != "05" => Messages(s"tcsp.service.provider.lbl.${provider.value}")
+        }
+      )
+      specialCase <- Some(providers.collect {
+          case provider if provider.value == "05" => Messages(s"tcsp.service.provider.lbl.05")
```

Comment:
Optional readability suggestion - could we use `provider.isInstanceOf[CompanyFormationAgent]` instead of `provider.value == "05"`? (Normally would avoid isInstanceOf, but ServiceProvider is a sealed trait...)

Reply:
good point @owen-opencast

Reply:
refactored